pymc/distributions/multivariate.py

```diff
@@ -2205,20 +2202,17 @@ def _infer_shape(self, size, dist_params, param_shapes=None):
 
         size = tuple(size)
 
-        return size + (K + 1,)
+        return size + tuple(alpha.shape) + (K + 1,)
```

Comment:
This is incorrect. Shape is always `size + (K + 1,), if size is provided. If size is not provided, the shape of alpha determines it. 

However a simpler solution is to remove this method and implement only `supp_shape_from_params` which should always return `(k + 1)` Aesara will then automatically know how then dimensions change with batched alpha or size info